analyses/survival-analysis/survival-analysis_tp53_telomerase.Rmd

```diff
@@ -0,0 +1,333 @@
+
+# combine tp53 scores and telomerase score to the dataframe
+# note that we do not need to recode categories since the survival_models.R functions will take care of it
+meta_indep <- histologies_rna %>%
+  # add tp53 and telomerase scores by matching Kids_First_Biospecimen_ID_RNA
+  left_join(tp53_scores, by = "Kids_First_Biospecimen_ID_RNA") %>%
+  left_join(tel_scores, by = "Kids_First_Biospecimen_ID_RNA") %>% 
+  arrange(Kids_First_Biospecimen_ID_RNA) %>%
+  distinct(Kids_First_Participant_ID, .keep_all=TRUE) %>%
+  # remove samples without survival data
+  drop_na(OS_days) %>%
+  # convert to years
+  mutate(OS_years = OS_days / 365.25)
+
+# Show the data we will use for analysis
+meta_indep
+```
+ 
+#### Create subsets of data for different models
+
+We create two new tibbles from `meta_indep`:
+
++ `meta_indep_hgg` which contains a new column `hgg_group` that contains levels (in order) `non-HGAT` and `HGAT` indicating whether it's an HGG sample
++ `meta_indep_cancers` which is subsetted to only certain cancer groups of interest that have suitable sample sizes which we find here.
+
+```{r hgg_split}
+# Define HGG groups
+hgg_cancer_groups <- c("High-grade glioma astrocytoma", "Diffuse midline glioma", "Diffuse intrinsic pontine glioma") 
+hgg_levels <- c("non-HGAT", "HGAT")
+
+# separate into HGG or non-HGG and make the new column a factor
+meta_indep_hgg <- meta_indep %>% 
+   mutate(hgg_group = ifelse(cancer_group %in% hgg_cancer_groups,
+                             hgg_levels[2], # second is HGAT 
+                             hgg_levels[1]), # first is non-HGAT
+          hgg_group = factor(hgg_group, levels = hgg_levels) # ensures non-HGAT is the reference level
+   )
+```
+
+```{r find_cancer_groups_to_use}
+# Define cancer groups of interest with higher Ns
+# We want to only include groups that have at least THREE in EITHER category. 
+# Find them here:
+cancer_groups_of_interest_df <- meta_indep %>%
+  drop_na() %>%
+  count(cancer_group,OS_status) %>%
+  spread(OS_status, n,
+         # any missing values (indicating 0 deceased or 0 living) should be 0, not NA
+         fill = 0) %>%
+  filter(DECEASED >= 3) 
+# show result: there are 7 groups to keep
+cancer_groups_of_interest_df
+
+# Subset to only the cancer groups with potentially suitable sample size
+meta_indep_cancer_groups <- meta_indep %>%
+  filter(cancer_group %in% cancer_groups_of_interest_df$cancer_group) %>%
+  # set reference level to Medulloblastoma
+  mutate(cancer_group = fct_relevel(cancer_group, "Medulloblastoma"))
+
+# How many patients got removed?
+nrow(meta_indep) - nrow(meta_indep_cancer_groups)
+```
+
+## Univariate Models
+
+ 
+Here, we run a survival analyses for several _independent_ predictors, one at a time. We use cox regression for all analyses (including categorical), as demonstrated in `survival-analysis_template.Rmd`.
+
++ `tp53_score` (continuous)
++ `telomerase_score` (continuous)
++ `hgg_group` (categorical)
++ `cancer_group` (categorical)
```

Comment:
One thing we also might want to do just as a gut check is run a univariate analysis on molecular subtype for `hgg_group` and `cancer_group` here which have subtypes. This can be in this notebook or can be a part of the immune deconv updates..

Reply:
I think since the immune deconv is already looking at molecular signatures an extent, we can do it there.

Reply:
sounds good!